doc/languages-frameworks/python.section.md

```diff
@@ -1983,14 +1985,24 @@ The following rules are desired to be respected:
 * Python applications live outside of `python-packages.nix` and are packaged
   with [`buildPythonApplication`](#buildpythonapplication-function).
 * Make sure libraries build for all Python interpreters.
-* By default we enable tests. Make sure the tests are found and, in the case of
+  If it fails to build on some Python versions, consider disabling them by setting `disable = pythonAtLeast "3.x"` along with a comment.
+* The two parameters, `pyproject` and `build-system` are set to avoid the legacy setuptools/distutils build.
+* Packages with versioning like `pydantic_1` are not included in the `dependencies`.
+  Such packages should not be used in libraries, because propagation causes conflicts.
```

Comment:
I think the takeaway could be something like this:

Only unversioned attributes (e.g. `pydantic`, but not `pypdantic_1`)  can be included in `dependencies`, since due to `PYTHONPATH` limitations we can only ever support a single version for libraries without running into duplicate module name conflicts.